test/junit/scala/tools/nsc/backend/jvm/BytecodeTest.scala

```diff
@@ -204,4 +204,21 @@ class BytecodeTest extends BytecodeTesting {
     val c = compileClass(code)
     assertEquals(c.attrs.asScala.toList.map(_.`type`).sorted, List("ScalaInlineInfo", "ScalaSig"))
   }
+
+  @Test
+  def t10343(): Unit = {
+    val main = "class Main { Person() }"
+    val person = "case class Person(age: Int = 1)"
+
+    def check(code: String) = {
+      val List(_, _, pm) = compileClasses(code)
+      assertEquals(pm.name, "Person$")
+      assertEquals(pm.methods.asScala.map(_.name).toList,
+        // after typer, `"$lessinit$greater$default$1"` is next to `<init>`, but the constructor phase
+        // and code gen change module constructors around. the second `apply` is a bridge, created in erasure.
+        List("<clinit>", "$lessinit$greater$default$1", "toString", "apply", "apply$default$1", "unapply", "readResolve", "apply", "<init>"))
```

Comment:
Side note: when creating the names of lambda impl methods, we opted for "new$" rather than "$lessinit$greater" (taking a leaf from the book of javac)